source/common/http/header_map_impl.cc

```diff
@@ -292,6 +292,9 @@ HeaderMapImpl::HeaderMapImpl(const HeaderMap& rhs) : HeaderMapImpl() {
       this);
 }
 
+HeaderMapImpl::HeaderMapImpl(const HeaderMapImpl& rhs)
+    : HeaderMapImpl(static_cast<const HeaderMap&>(rhs)) {}
```

Comment:
Sorry I'm confused. How is this different from what the compiler did before?

Reply:
It was synthesizing the default copy constructor, since we only supplied a constructor for the class we were deriving from AFAICT.

Reply:
There's something more going on here though, witness the build failures. Looking into this now.